File: kotlin/src/main/kotlin/Claass204/RPS.kt
package Claass204

import java.util.Scanner
import kotlin.random.Random

fun main() {
    val scanner = Scanner(System.`in`)
    print("R, P, or S: ")
    val user = scanner.nextLine().trim().toUpperCase()
    val rpsTable = listOf("R", "P", "S")
    val bot = rpsTable[Random.nextInt(rpsTable.size)]
    val result = rpsTable.indexOf(user) - rpsTable.indexOf(bot)

    when {
        result == 0 -> println("$user vs $bot Draw")
        result == 1 || result == -2 -> println("$user vs $bot You Win")
        else -> println("$user vs $bot You Lose")
    }
}

File: Lua/al/RPS.lua
function table.indexOf(t, value)
    for i, v in ipairs(t) do
        if v == value then
            return i
        end
    end
    return nil
end
for i = 1, 3 do
    io.write("R, P, or S: ")
    user = io.read()

    rps_table = {"R", "P", "S"}

    bot = rps_table[math.random(#rps_table)]

    result = table.indexOf(rps_table, user) - table.indexOf(rps_table, bot)

    if result == 0 then
        print(string.format('%s vs %s  Draw', user, bot))
    elseif result == 1 or result == -2 then
        print(string.format('%s vs %s  You Wins', user, bot))
    else
        print(string.format('%s vs %s  You Lose', user, bot))
    end
end